akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -142,14 +142,27 @@ private[akka] final case class Take[T](count: Long) extends PushPullStage[T, T]
 /**
  * INTERNAL API
  */
-private[akka] final case class Drop[T](count: Long) extends PushStage[T, T] {
-  private var left: Long = count
+private[akka] final case class Drop[T](count: Long) extends SimpleLinearGraphStage[T] {
+  override def initialAttributes: Attributes = DefaultAttributes.drop
 
-  override def onPush(elem: T, ctx: Context[T]): SyncDirective =
-    if (left > 0) {
-      left -= 1
-      ctx.pull()
-    } else ctx.push(elem)
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new GraphStageLogic(shape) {
+    private var left: Long = count
+
+    setHandler(in, new InHandler {
```

Comment:
You can save two allocations here by implementing the `InHandler` and `OutHandler` with in the `GraphStageLogic` itself, and use that `setHandlers(in, out, this)`